### ISSUE TITLE
Introduce a Ballerina connector for Mistral REST API

### ISSUE THREAD

**Issue by NipunaRanasinghe** [ballerina-platform/ballerina-library#7046]
We are planning to introduce a new Ballerina connector for the latest Mistral REST API, generated using it's OpenAPI specification.

Related links:
- https://docs.mistral.ai/api/
- https://github.com/mistralai/platform-docs-public/blob/main/openapi.yaml

This includes the following tasks:
1. Create a new repository under 'ballerina-platform' organization.
2. Review and sanitize the OpenAPI specification.
3. Generate the connector client.
4. Implement a test suite covering the core functionalities.
5. Write documentation with examples.

**Comment by someshfengde** [comment]
Hi @NipunaRanasinghe  can you please assign this issue to me 

thanks :)

**Comment by someshfengde** [comment]
Hi @keizer619  don't have access to create a repo under ballerina-platform org?

**Comment by someshfengde** [comment]
Also can you please provide me an example of existing connector client?

**Comment by someshfengde** [comment]
@keizer619  is there any update on this

**Comment by Nuvindu** [comment]
👋 Welcome, @someshfengde ! 🚀

We're thrilled to have you join the [Ballerina Hacktoberfest](https://ballerina.io/hacktoberfest/) community! The issue has been assigned to you, and we’re excited to see your contributions.

To help you get started, here are a few essential resources:

1. **Connector Contributor Guide**: Make sure to go through our [Ballerina Hacktoberfest connector contributor guidelines](https://ballerina.io/hacktoberfest/connector-contributor-guide/) and follow the exact steps, to contribute effectively.
2. **Learn Ballerina**: Whether you're new to Ballerina or looking to enhance your skills, check out our [official learning resources](https://ballerina.io/learn/).
3. **Community & Support**: If you have technical questions, feel free to ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/ballerina) with the Ballerina tag, or join us on [Discord](https://discord.com/invite/ballerinalang) to connect with other community members.

No contribution is too small, and your feedback is always welcome! Don’t hesitate to ask questions, propose new ideas, or report issues.

We are currently in the process of creating a GitHub repository for this connector module and will update you once it’s available. Therefore you don't have to create the repository yourself. In the meantime, please go through the relevant resources and documentation related to the connector.

**Happy coding** and welcome aboard! 🎉

**Comment by Nuvindu** [comment]
> Also can you please provide me an example of existing connector client?

Here, you can find the Ballerina connector module for Gmail APIs :
https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail

**Comment by AzeemMuzammil** [comment]
Hi @someshfengde,

I'll be mentoring your project, and we have created the repository for your connector project with the initial project template. You can start working in this repository: [module-ballerinax-mistral](https://github.com/ballerina-platform/module-ballerinax-mistral).

Please refer to the contribution process section for guidelines on raising a PR: https://ballerina.io/hacktoberfest/connector-contributor-guide/#contribution-process.

Happy coding! 🎉

**Comment by AzeemMuzammil** [comment]
Hi @someshfengde, have you had a chance to start working on this issue? Please let me know if you're facing any challenges or need any assistance.

**Comment by AzeemMuzammil** [comment]
As we haven't received any updates from @someshfengde  in the past week, we are removing the assignment for now. If you are still interested in working on this issue, please comment below. Meanwhile, this issue is now open for contributions from others.

**Comment by ayeshLK** [comment]
@AzeemMuzammil, @thiunuwan began working on the Ballerina SmartSheet connector, but due to limitations in the OpenAPI tool, we weren’t able to proceed with development. If there isn’t an assignee for the Ballerina Mistral connector, he is available and willing to take it on.

**Comment by thiunuwan** [comment]
> @AzeemMuzammil, @thiunuwan began working on the Ballerina SmartSheet connector, but due to limitations in the OpenAPI tool, we weren’t able to proceed with development. If there isn’t an assignee for the Ballerina Mistral connector, he is available and willing to take it on.

 Yes,I would like to work on this issue, can you please assign this to me?

**Comment by ayeshLK** [comment]
Greetings from Ballerina team @thiunuwan 👋 

You can find the repository for the Ballerina Mistral connector here [1]

To provide context, the deadline for the development is established as one month from the date you were officially assigned to the project. But if you encounter any challenges or unforeseen issues, please feel free to reach out, and we can consider adjusting the deadline accordingly.

If you have any questions or need help with anything, please feel free to reach out. Keep up the awesome work! 🚀

[1] - https://github.com/ballerina-platform/module-ballerinax-mistral

**Comment by ayeshLK** [comment]
Hey @thiunuwan 👋 

Just checking in on the progress of the connector development.

Were you able to generate the connector using the official OpenAPI specification [1][2] ? 

If you have any concerns regarding the connector development, please don't hesitate to reach out.

Happy Coding 🚀 

[1] - https://docs.mistral.ai/api/
[2] - https://github.com/mistralai/platform-docs-public/blob/main/openapi.yaml

**Comment by ayeshLK** [comment]
@thiunuwan 👋 

Just checking in on the progress. Were you able to generate the client code using the OpenAPI specification ? If so can you send the pull-request so that I could help you by reviewing it.

**Comment by thiunuwan** [comment]
> @thiunuwan 👋
> 
> Just checking in on the progress. Were you able to generate the client code using the OpenAPI specification ? If so can you send the pull-request so that I could help you by reviewing it.

Hi @ayeshLK ,
Sorry for the delayed response—I haven’t been feeling well past few days.I was able to generate the client code. I have to write the necessary tests next, then send a pull request for your review

**Comment by ayeshLK** [comment]
Hey @thiunuwan 

Sorry to hear that you were not well. Hope you are feeling well now. 

> I have to write the necessary tests next, then send a pull request for your review

Yes, that would be great. Please let me know if you need any additional assistance.

**Comment by ayeshLK** [comment]
> Hi @ayeshLK , I have submitted a PR for the client code generation and have also set up the environment for testing. Also, could you please confirm if the sample test I have written is correct? Looking forward to your feedback.

@thiunuwan great, I will review it ASAP

**Comment by ayeshLK** [comment]
Hey @thiunuwan 👋 

I have added set of review comments to your PR. Can you please check ?

**Comment by thiunuwan** [comment]
> I have added set of review comments to your PR. Can you please check ?

Sure, I’ll check them.

**Comment by ayeshLK** [comment]
@thiunuwan I have added few more minor comments to your PR. Could you please check and resolve them ?

**Comment by thiunuwan** [comment]
> @thiunuwan I have added few more minor comments to your PR. Could you please check and resolve them ?

Hi @ayeshLK ,
I've resolved the minor issues you mentioned.
I will add more test cases as u mentioned. ( I just wanted to check whether the way I wrote the test cases is correct, which is why I added only one test case. )

**Comment by ayeshLK** [comment]
> Hi @ayeshLK , I've resolved the minor issues you mentioned. Also, I will add more test cases as you mentioned. ( I just wanted to check whether the way I wrote the test cases is correct, which is why I added only one test case. )

@thiunuwan the way you have written the tests are correct. You can follow the same approach to add few more test cases.

**Comment by ayeshLK** [comment]
@thiunuwan are you still working on this connector ? It seems like we are gone past the deadline (1st December) as we have initially assigned this issue to you on 1st November. Please check this comment [1] for more details.

If you need more time could you please let us know ?

[1] - https://github.com/ballerina-platform/ballerina-library/issues/7046#issuecomment-2451254480

**Comment by thiunuwan** [comment]
> @thiunuwan are you still working on this connector ? It seems like we are gone past the deadline (1st December) as we have initially assigned this issue to you on 1st November. Please check this comment [1] for more details.
> 
> If you need more time could you please let us know ?
> 
> [1] - [#7046 (comment)](https://github.com/ballerina-platform/ballerina-library/issues/7046#issuecomment-2451254480)

Hi @ayeshLK  , 
Apologies for not being able to complete this on time. I've been a bit busy with my academic work and couldn't dedicate enough time to focus on this issue. If someone else is interested, please feel free to assign it to them.
If the issue remains unresolved by the time I am available, I will complete the remaining work.

**Comment by keizer619** [comment]
> > [@thiunuwan](https://github.com/thiunuwan) are you still working on this connector ? It seems like we are gone past the deadline (1st December) as we have initially assigned this issue to you on 1st November. Please check this comment [1] for more details.
> > If you need more time could you please let us know ?
> > [1] - [#7046 (comment)](https://github.com/ballerina-platform/ballerina-library/issues/7046#issuecomment-2451254480)
> 
> Hi [@ayeshLK](https://github.com/ayeshLK) , Apologies for not being able to complete this on time. I've been a bit busy with my academic work and couldn't dedicate enough time to focus on this issue. If someone else is interested, please feel free to assign it to them. If the issue remains unresolved by the time I am available, I will complete the remaining work.

Hi @thiunuwan 
Since you already engaged and If you are willing to work you can continue the work.

Now we will be rewarding these community contributions beyond the Hacktoberfest. Please refer the [website for more information](https://ballerina.io/community/#contribute-and-get-rewarded)

**Comment by anupama-pathirage** [comment]
@thiunuwan Just checking in — we haven’t seen any updates on this issue for a while. To keep things moving, we’ll go ahead and remove the current assignment so that others in the community can take a look.

If you're still interested in working on this, feel free to comment here and we’ll be happy to reassign it to you. Thanks for your interest in contributing! 🙌

**Comment by daneshk** [comment]
The Mistral.ai connector is released as part of the AI effort. Hence, closing the issue.
